main/SQLHelper.cpp

```diff
@@ -6017,9 +6066,32 @@ void CSQLHelper::UpdateTemperatureLog()
 			{
 			case pTypeRego6XXTemp:
 			case pTypeTEMP:
-			case pTypeThermostat:
 				temp = static_cast<float>(atof(splitresults[0].c_str()));
 				break;
+			case pTypeThermostat:
+				{
+
+					const std::string& HwdId = sd[7];
+					//get virtual thermostat hardware
+					CDomoticzHardwareBase* Hardware = m_mainworker.GetHardwareByIDType( HwdId, HTYPE_VirtualThermostat);
+					//if not a virtual thermostat
+					if (Hardware==nullptr)
+					{
+						temp = static_cast<float>(std::stof(splitresults[0].c_str()));
```

Comment:
you do not need .c_str() here, also the cast to float is not needed (std::stof is returning a float)

Reply:
Ok done.